src/main/kotlin/org/rust/lang/core/types/infer/TypeInference.kt

```diff
@@ -169,48 +141,66 @@ private class RsFnInferenceContext(
     }
 
     private fun inferStructLiteralType(expr: RsStructLiteral): Ty {
-        val (element, subst) = expr.path.reference.advancedResolve() ?: return TyUnknown
+        val boundElement = expr.path.reference.advancedResolve()
+        if (boundElement == null) {
+            inferStructTypeArguments(expr)
+            return TyUnknown
+        }
+        val (element, subst) = boundElement
         return when (element) {
             is RsStructItem -> element.type
-                .substitute(subst)
-                .substitute(inferStructTypeArguments(expr))
             is RsEnumVariant -> element.parentEnum.type
-                .substitute(subst)
-                .substitute(inferStructTypeArguments(expr))
             else -> TyUnknown
-        }
+        }.substitute(subst).substitute(inferStructTypeArguments(expr))
+    }
+
+    private fun inferRsTupleExprType(expr: RsTupleExpr): Ty {
+        return TyTuple(expr.exprList.map { inferExprType(it) })
+    }
+
+    fun inferCastExprType(expr: RsCastExpr): Ty {
```

Comment:
Looks like private is missing here, so even GH highlighting is confused :) And some other functions probably miss private as well.